isis/tests/PixelTests.cpp

```diff
@@ -0,0 +1,481 @@
+  // HIS
+  EXPECT_EQ(Isis::HIGH_INSTR_SAT2, Pixel(1, 2, 3, Isis::His).To16Bit());
+  // LRS
+  EXPECT_EQ(Isis::LOW_REPR_SAT2, Pixel(1, 2, 3, Isis::Lrs).To16Bit());
+  // LIS
+  EXPECT_EQ(Isis::LOW_INSTR_SAT2, Pixel(1, 2, 3, Isis::Lis).To16Bit());
+}
+
+TEST(Pixel, static_To32Bit) {
+  // Zero test
+  EXPECT_EQ(0, Pixel::To32Bit(0.0));
+  // Negative test
+  EXPECT_EQ(-1, Pixel::To32Bit(-1.0));
+  // Positive test
+  EXPECT_EQ(1, Pixel::To32Bit(1.0));
+  // Minimum valid input
+  EXPECT_FLOAT_EQ(Isis::LOW_REPR_SAT4, Pixel::To32Bit(Isis::ValidMinimum));
+  // Maximum valid input // Isis::Maximum becomes inf (not IHIGH_REPR_SAT4)
+  // EXPECT_FLOAT_EQ(Isis::HIGH_REPR_SAT4, Pixel::To32Bit(Isis::ValidMaximum));
+  // "Null" pixel
+  EXPECT_FLOAT_EQ(Isis::NULL4, Pixel::To32Bit(Isis::Null));
+  // HRS
+  EXPECT_FLOAT_EQ(Isis::HIGH_REPR_SAT4, Pixel::To32Bit(Isis::Hrs));
+  // HIS
+  EXPECT_FLOAT_EQ(Isis::HIGH_INSTR_SAT4, Pixel::To32Bit(Isis::His));
+  // LRS
+  EXPECT_FLOAT_EQ(Isis::LOW_REPR_SAT4, Pixel::To32Bit(Isis::Lrs));
+  // LIS
+  EXPECT_FLOAT_EQ(Isis::LOW_INSTR_SAT4, Pixel::To32Bit(Isis::Lis));
+}
+
+TEST(Pixel, To32Bit) {
+  // Zero test
+  EXPECT_EQ(0, Pixel(1, 2, 3, 0.0).To32Bit());
+  // Negative test
+  EXPECT_EQ(-1, Pixel(1, 2, 3, -1.0).To32Bit());
+  // Positive test
+  EXPECT_EQ(1, Pixel(1, 2, 3, 1.0).To32Bit());
+  // Minimum valid input
+  EXPECT_FLOAT_EQ(Isis::LOW_REPR_SAT4, Pixel(1, 2, 3, Isis::ValidMinimum).To32Bit());
+  // Maximum valid input // Isis::Maximum becomes inf (not HIGH_REPR_SAT4)
+  // EXPECT_FLOAT_EQ(Isis::HIGH_REPR_SAT4, Pixel(1, 2, 3, Isis::ValidMaximum).To32Bit());
+  // "Null" pixel
+  EXPECT_FLOAT_EQ(Isis::NULL4, Pixel(1, 2, 3, Isis::Null).To32Bit());
+  // HRS
+  EXPECT_FLOAT_EQ(Isis::HIGH_REPR_SAT4, Pixel(1, 2, 3, Isis::Hrs).To32Bit());
+  // HIS
+  EXPECT_FLOAT_EQ(Isis::HIGH_INSTR_SAT4, Pixel(1, 2, 3, Isis::His).To32Bit());
+  // LRS
+  EXPECT_FLOAT_EQ(Isis::LOW_REPR_SAT4, Pixel(1, 2, 3, Isis::Lrs).To32Bit());
+  // LIS
+  EXPECT_FLOAT_EQ(Isis::LOW_INSTR_SAT4, Pixel(1, 2, 3, Isis::Lis).To32Bit());
+}
+
+TEST(Pixel, static_ToDouble) {
+  // unsigned char
+  unsigned char uc = 0;
+  EXPECT_DOUBLE_EQ(Isis::Null, Pixel::ToDouble(uc));
+
+  // short
+  short s = 0.0;
+  EXPECT_DOUBLE_EQ(0.0, Pixel::ToDouble(s));
+
+  // unsigned short
+  unsigned short us = 0.0;
+  EXPECT_DOUBLE_EQ(Isis::Null, Pixel::ToDouble(us));
+
+  // float
+  float f = 0.0;
+  EXPECT_DOUBLE_EQ(0.0, Pixel::ToDouble(f));
+}
+
+TEST(Pixel, ToDouble) {
+  EXPECT_EQ(0.0, Pixel(1, 2, 3, 0.0).ToDouble());
+}
+
+TEST(Pixel, static_ToFloat) {
+  // unsigned char
+  unsigned char uc = 0;
+  EXPECT_DOUBLE_EQ(Isis::NULL4, Pixel::ToFloat(uc));
+
+  // short
+  short s = 0.0;
+  EXPECT_DOUBLE_EQ(0.0, Pixel::ToFloat(s));
+
+  // unsigned short
+  unsigned short us = 0.0;
+  EXPECT_DOUBLE_EQ(Isis::NULL4, Pixel::ToFloat(us));
+
+  // float
+  float f = 0.0;
+  EXPECT_DOUBLE_EQ(0.0, Pixel::ToFloat(f));
```

Comment:
These should use either EXPECT_EQ or EXPECT_FLOAT_EQ instead of EXPECT_DOUBLE_EQ because these are actually 32 bit floats.

Reply:
right, i went ahead and added EXPECT_FLOAT_EQ